sources/Algo/include/Algo.h

```diff
@@ -323,20 +323,49 @@ struct HvdcLineDefinition {
   const ConverterId converter2_id;                             ///< second converter id
   const BusId converter2_busId;                                ///< second converter bus id
   const boost::optional<bool> converter2_voltageRegulationOn;  ///< second converter voltage regulation parameter, for VSC converters only
-  const Position position;                                     ///< position of the converters of this hvdc line compared to the main connex component
+  mutable Position position;                                   ///< position of the converters of this hvdc line compared to the main connex component
 };
 
+/**
+   * @brief Hash structure for term
+   *
+   * required to use set on terms
+   */
+struct HashTerm {
+  /**
+   * @brief Hash function
+   *
+   * @param hvdcLine the term to hash
+   * @returns the hash value of @p hvdcLine
+   */
+  std::size_t operator()(const HvdcLineDefinition& hvdcLine) const {
+    return std::hash<std::string>{}(hvdcLine.id);
+  }
+  /**
+   * @brief Less function
+   *
+   * @param lhs the left hand side to compare
+   * @param rhs the right hand side to compare
+   * @returns true if lhs is inferior to rhs based on their id
+   */
+  bool operator()(const HvdcLineDefinition& lhs, const HvdcLineDefinition& rhs) const {
```

Comment:
this function should not be in the hash struct. It is the comparator function. Remove if not different from the already defined '<' operator